anitya/lib/backends/npmjs.py

```diff
@@ -92,3 +92,28 @@ def get_versions(cls, project):
             raise AnityaPluginException('No versions found at %s' % url)
 
         return data['versions'].keys()
+
+    @classmethod
+    def check_feed(cls):
+        ''' Return a generator over the latest 40 uploads to npmjs.org
+
+        by querying an weird JSON endpoint.
+        '''
+
+        url = 'https://registry.npmjs.org/-/all/static/today.json'
+
+        try:
+            response = cls.call_url(url)
+        except Exception:  # pragma: no cover
+            raise AnityaPluginException('Could not contact %s' % url)
+
+        try:
+            data = response.json()
+        except Exception:  # pragma: no cover
+            raise AnityaPluginException('No JSON returned by %s' % url)
+
+        for item in data[:40]:
```

Comment:
"full scan"? There is nothing like full scan as far as I understand. The "full scan" looks just for known packages, i.e. if you miss newly introduced package, the record for that package is never created in Anitya and hence it won't be discovered until its new release is caught by this code again.